### PR TITLE
mediatek-scaler: Fix 'structurally dead code' spotted by Coverity

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by CragW — APPROVED** [review]

File: plugins/mediatek-scaler/fu-mediatek-scaler-device.c [hunk 1]
```diff
@@ -127,6 +127,8 @@ fu_mediatek_scaler_device_use_aux_dev(FuDevice *device, GError **error)
 
 		udev = fu_udev_device_new(fu_device_get_context(device), element->data);
 		i2c_devices = fu_udev_device_get_siblings_with_subsystem(udev, "i2c");
+		if (i2c_devices->len == 0)
+			continue;
```
Comment (CragW) [inline plugins/mediatek-scaler/fu-mediatek-scaler-device.c:131]: I thought it eventually report the same error as current flow is?

Comment (hughsie) [reply plugins/mediatek-scaler/fu-mediatek-scaler-device.c:131]: I think what coverity is saying is "you're always choosing the first device, why bother with the loop"

Comment (hughsie) [reply plugins/mediatek-scaler/fu-mediatek-scaler-device.c]: So do you think my first patch is appropriate, or should we do something like the fix-up instead?

Comment (CragW) [reply plugins/mediatek-scaler/fu-mediatek-scaler-device.c]: Yes, you are correct. We actually don't need iterating the dp_aux devices again. I've sent another patch if that change makes sensible to you.

Comment (hughsie) [reply plugins/mediatek-scaler/fu-mediatek-scaler-device.c]: Better indeed! Thanks.

**Pull request merged**